### ISSUE TITLE
Light Managers constructed at the wrong time

### ISSUE THREAD

**Issue by jamesu** [GarageGames/Torque3D#733]
In Torque3D, BasicLightManager is created in a module, i.e. 

```
MODULE_BEGIN( BasicLightManager )
   MODULE_SHUTDOWN_AFTER( Scene )
   MODULE_INIT
   {
      ManagedSingleton< BasicLightManager >::createSingleton();
   }
   MODULE_SHUTDOWN
   {
      ManagedSingleton< BasicLightManager >::deleteSingleton();
   }
MODULE_END;
```

Simple enough. However this creates a problem with the light fields for each light object which are created via `LightManager::initLightFields()` (called from ::initPersistFields). This function iterates through each light manager and adds their fields to the object. However when ::initPersistFields is called, the module system hasn't been initialzed yet thus no fields will be added for BasicLightManager.

AdvancedLightManager doesn't have this problem since it's initialized on startup outside of the module system. However this introduces another problem: depending on the order in which AdvancedLightManager is destroyed, crashes can occur due to either memory corruption or invalid objects.

There is also an additional problem with BasicLightManager if you try to "fix" the problem using the AdvancedLightManager approach. In the constructor, it registers several console variables however this only works when the console system has been initialized, not before.

In short, the initialization order might need a rethink.
